src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -29,6 +32,8 @@ internal sealed partial class VisualStudioProjectTracker : ForegroundThreadAffin
         #region Mutable fields accessed only from foreground thread - don't need locking for access (all accessing methods must have AssertIsForeground).
         private readonly List<WorkspaceHostState> _workspaceHosts;
 
+        private readonly HostWorkspaceServices _workspaceServices;
```

Comment:
You can probably just pass in the `ICommandLineParserService` when we initialize other providers/services for the project tracker: http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs,89